File: app/models/address.rb
# == Schema Information
#
# Table name: addresses
#
#  id         :bigint           not null, primary key
#  name       :string
#  street1    :string
#  street2    :string
#  city_id    :bigint           not null
#  state      :string
#  zip_code   :string
#  created_at :datetime         not null
#  updated_at :datetime         not null
#
class Address < ApplicationRecord
  belongs_to :city
  has_many :farmers_markets
end

File: src/VegDex.Core/Entities/Address.cs
using VegDex.Core.Entities.Base;

namespace VegDex.Core.Entities;

public class Address : Entity
{
    public Address() { }
    public City City { get; set; }
    public string? Name { get; set; }
    public string? State { get; set; }
    public string? Street1 { get; set; }
    public string? Street2 { get; set; }
    public string? ZipCode { get; set; }
    public static Address Create(
        int addressId,
        City city,
        string? name = null,
        string? street1 = null,
        string? street2 = null,
        string? state = null,
        string? zipCode = null)
    {
        var address = new Address
        {
            Id = addressId,
            Name = name,
            City = city,
            Street1 = street1,
            Street2 = street2,
            State = state,
            ZipCode = zipCode
        };
        return address;
    }
}
